packages/vue-apollo-composable/src/useApolloClient.ts

```diff
@@ -37,9 +37,12 @@ export function useApolloClient<TCacheShape = any>(clientId?: ClientId): UseApol
   const providedApolloClients: ClientDict<TCacheShape> = inject(ApolloClients, null)
   const providedApolloClient: ApolloClient<TCacheShape> = inject(DefaultApolloClient, null)
 
-  function resolveClient() {
+  function resolveClient(resolvedClientId?: ClientId) {
+    if (resolvedClientId) {
+      clientId = resolvedClientId
```

Comment:
Doesn't this need a local variable to avoid overriding this for all subsequent requests?

Reply:
Yes, you are right